src/Ordering.API/Apis/OrdersApi.cs

```diff
@@ -135,9 +137,10 @@ public static async Task<OrderDraftDTO> CreateOrderDraftAsync(CreateOrderDraftCo
 
         using (services.Logger.BeginScope(new List<KeyValuePair<string, object>> { new("IdentifiedCommandId", requestId) }))
         {
+            var maskedCCNumber = request.CardNumber.Replace(request.CardNumber.Substring(0, 12), new string('X', 12));
```

Comment:
I have no idea what is more efficient for the compiler, but allocating a new string may not be? I would have used PadLeft but maybe someone tells me that's bad.


`request.CardNumber.Substring(request.CardNumber.Length - 4).PadLeft(request.CardNumber.Length, 'X');`

Reply:
Given that it's an order, which isn't a high volume thing, is this really an issue?